src/difference.rs

```diff
@@ -0,0 +1,114 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use crate::{digraph, graph, StablePyGraph};
+
+use hashbrown::HashSet;
+
+use petgraph::visit::{EdgeRef, IntoEdgeReferences};
+use petgraph::{algo, EdgeType};
+
+use pyo3::exceptions::PyIndexError;
+use pyo3::prelude::*;
+use pyo3::Python;
+
+fn difference<Ty: EdgeType>(
+    py: Python,
+    first: &StablePyGraph<Ty>,
+    second: &StablePyGraph<Ty>,
+) -> PyResult<StablePyGraph<Ty>> {
+    let indexes_first = first.node_indices().collect::<HashSet<_>>();
+    let indexes_second = second.node_indices().collect::<HashSet<_>>();
+
+    if indexes_first != indexes_second {
+        return Err(PyIndexError::new_err(
+            "Node sets of the graphs should be equal",
+        ));
+    }
+
+    let mut final_graph = StablePyGraph::<Ty>::with_capacity(
+        first.node_count(),
+        first.edge_count() - second.edge_count(),
+    );
+
+    for node in first.node_indices() {
+        let weight = &first[node];
+        final_graph.add_node(weight.clone_ref(py));
+    }
```

Comment:
What happens if the node weights are different between first and second? Do we care at all that if `first[0] != second[0]` we only preserve the payload for `first[0]`?

Reply:
Do you think we should have a callback to handle weights from both graphs?